File: Quivi.Pos.Api/Dtos/Requests/Orders/GetOrdersRequest.cs
using Quivi.Domain.Entities.Pos;
using Quivi.Infrastructure.Abstractions.Repositories.Data;

namespace Quivi.Pos.Api.Dtos.Requests.Orders
{
    public class GetOrdersRequest : APagedRequest
    {
        public IEnumerable<string>? Ids { get; init; }
        public IEnumerable<string>? ChannelIds { get; init; }
        public IEnumerable<string>? SessionIds { get; init; }
        public IEnumerable<OrderState>? States { get; init; }
        public SortDirection? SortDirection { get; init; }
    }
}

File: quivi.pos.react.app/src/hooks/api/Dtos/orders/GetOrdersRequest.ts
import { PagedRequest } from "../PagedRequest";
import { SortableRequest } from "../SortableRequest";
import { OrderState } from "./OrderState";

export interface GetOrdersRequest extends PagedRequest, SortableRequest {
    readonly ids?: string[];
    readonly states?: OrderState[];
    readonly channelIds?: string[];
    readonly sessionIds?: string[];
}